solutions/cf-862B.mdx

```diff
@@ -0,0 +1,122 @@
+---
+id: cf-862B
+source: CF
+title: Bipartiteness
+author: Ryan Chou
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/54590) 
+
+## Explanation
+Given the number of colored and uncolored nodes, the maximum number of edges that can exist is $c \cdot uc$ where $c$ represents the number of colored nodes, and $uc$ represents the number of uncolored nodes. 
+
+This is because an edge can be drawn from each colored node to every other uncolored node. However, we already have $n - 1$ edges, so the answer is $(c \cdot uc) - (n - 1)$.
+
+To find the number of colored and uncolored nodes, we can run a [DFS](https://usaco.guide/silver/dfs), incrementing the number of colored and uncolored nodes as we traverse the graph.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Bipartiteness {
+	public static ArrayList<Integer> adj[];
+	
+	// counters for colored and uncolored nodes
+	public static long[] cnt = {0, 0};
+
+	public static void main(String[] args) throws IOException{
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		PrintWriter out = new PrintWriter(System.out);
+
+		int n = Integer.parseInt(br.readLine());
+
+		// +1 for 1-indexed nodes
+		adj = new ArrayList[n + 1];
+		
+		for (int i = 0; i <= n; i++) adj[i] = new ArrayList<Integer>();
+
+		StringTokenizer s;
```

Comment:
why not put this in the for loop
does it make it tle or smth

Reply:
it's faster, does it make more sense if it is in the for loop?

Reply:
resolved by using kattio